service/controller/release/resource/status/cluster.go

```diff
@@ -0,0 +1,205 @@
+package status
+
+import (
+	"context"
+
+	apiexlabels "github.com/giantswarm/apiextensions/v2/pkg/label"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	azurecapi "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+
+	"github.com/giantswarm/release-operator/service/controller/key"
+)
+
+type TenantCluster struct {
+	ID               string
+	OperatorVersion  string
+	ProviderOperator string
+	ReleaseVersion   string
+}
+
+// Takes a list of tenant clusters and returns two maps containing the versions of their release and operator versions.
+func consolidateClusterVersions(clusters []TenantCluster) (map[string]bool, map[string]map[string]bool) {
```

Comment:
I don't really like returning multiple unnamed values like this. I think it would be better to pass the maps in as arguments, give them their own type (`type ReleaseUsageMap map[string]bool`), or use named return values.

Reply:
Sure, good idea. I gave them some names